tmk_core/avr.mk

```diff
@@ -171,7 +176,16 @@ dfu-ee: $(BUILD_DIR)/$(TARGET).hex $(BUILD_DIR)/$(TARGET).eep
 
 avrdude: $(BUILD_DIR)/$(TARGET).hex check-size cpfirmware
 	if $(GREP) -q -s Microsoft /proc/version; then \
-		echo 'ERROR: AVR flashing cannot be automated within the Windows Subsystem for Linux (WSL) currently. Instead, take the .hex file generated and flash it using AVRDUDE, AVRDUDESS, or XLoader.'; \
+		printf "Detecting USB port, reset your controller now."; \
+		COMPORT1=`$(MODECMD)|grep -o 'COM[0-9]*'` ;\
+		WINAVRPATH=`echo $(WINAVRDIR)|perl -pne 's#/mnt/c/#C:/#'|perl -pne 's#\E/#\\\#g'`; \
+		while [ -z $$COMPORT2 ]; do \
```

Comment:
The problem here is that there is absolutely no detection here.  It picks the FIRST available COM port.  This is great if you only have the one Pro Micro in bootloader mode, but if you have any other COM ports (such as arduino sketches, debugging tools, etc), then there is a good chance that this will error out, because it's using the wrong port. 

I tested this on my system (had to reflash caterina :D), and it picks the first COM port listed from `mode.com`.  Which is not correct (that COM port is an Arduino sketch I use for lighting up my desk)

So some sort of detection is needed here, like below.  To see what COM part has "changed" (eg just connected).